railties/lib/rails/generators/rails/app/app_generator.rb

```diff
@@ -244,11 +251,31 @@ def create_test_files
       end
 
       def create_tmp_files
-        build(:tmp)
+        build(:tmp) unless options[:api]
```

Comment:
Just had a discussion with @spastorino. We agree that there is no reason to remove `tmp` or `vendor` folder.

We will revert this change.